### PR TITLE
ddl: check foreign key constraint when drop,modify,change column (#14043)

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
cherry-pick #14043

parser  https://github.com/pingcap/parser/pull/690

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -75,3 +75,5 @@ require (
 )
 
 go 1.13
+
+replace github.com/pingcap/parser => github.com/crazycs520/parser v0.0.0-20191218032222-cb5d742ae9e0
```
Comment (wjhuang2016) [inline go.mod]: 。。。

File: go.sum [hunk 2]
```diff
@@ -24,6 +24,8 @@ github.com/coreos/go-systemd v0.0.0-20181031085051-9002847aa142/go.mod h1:F5haX7
 github.com/coreos/pkg v0.0.0-20160727233714-3ac0863d7acf/go.mod h1:E3G3o1h8I7cfcXa63jLwjI0eiQQMgzzUDFVpN/nH/eA=
 github.com/coreos/pkg v0.0.0-20180928190104-399ea9e2e55f h1:lBNOc5arjvs8E5mO2tbpBpLoyyu8B6e44T7hJy6potg=
 github.com/coreos/pkg v0.0.0-20180928190104-399ea9e2e55f/go.mod h1:E3G3o1h8I7cfcXa63jLwjI0eiQQMgzzUDFVpN/nH/eA=
+github.com/crazycs520/parser v0.0.0-20191218032222-cb5d742ae9e0 h1:3Z4NvkQhzgJsMhY1x7EVwh167Rk49Fwc1/8RoDQ8Yhc=
```
Comment (wjhuang2016) [inline go.sum]: Remove it

File: go.mod (hunk 1 above)
Comment (crazycs520) [reply go.mod]: https://github.com/pingcap/parser/pull/690 merge this first

File: ddl/db_test.go [hunk 3]
```diff
@@ -2053,7 +2053,20 @@ func (s *testDBSuite2) TestTableForeignKey(c *C) {
...
+	s.tk.MustExec("create table t4 (c int,d int,foreign key (d) references t1 (b));")
+	failSQL = "alter table t4 drop column d"
+	s.tk.MustGetErrCode(failSQL, mysql.ErrFkColumnCannotDrop)
+	// Test change column with foreign key.
+	failSQL = "alter table t4 change column d e bigint;"
+	s.tk.MustGetErrCode(failSQL, mysql.ErrFKIncompatibleColumns)
+	// Test modify column with foreign key.
+	failSQL = "alter table t4 modify column d bigint;"
+	s.tk.MustGetErrCode(failSQL, mysql.ErrFKIncompatibleColumns)
+	s.tk.MustQuery("select count(*) from information_schema.KEY_COLUMN_USAGE;")
```
Comment (Deardrops) [inline ddl/db_test.go]: check result == 0?

Comment (crazycs520) [reply ddl/db_test.go]: No need to do this, just test it won't panic.

**Pull request merged**